src/game/Spells/Spell.cpp

```diff
@@ -2437,17 +2440,49 @@ void Spell::SetTargetMap(SpellEffectIndex effIndex, uint32 targetMode, bool targ
                                 break;
                             }
                         }
-
                     }
                 }
-             
+
                 UnitList tempAoeList;
-                MaNGOS::AnyFriendlyOrGroupMemberUnitInUnitRangeCheck u_check(m_caster, group, m_spellInfo, max_range);
-                MaNGOS::UnitListSearcher<MaNGOS::AnyFriendlyOrGroupMemberUnitInUnitRangeCheck> searcher(tempAoeList, u_check);
-                Cell::VisitAllObjects(m_caster, searcher, max_range);
+                // Getting spell casting distance
+                float minRadiusCaster = 0, maxRadiusTarget = 0;
+                GetChainJumpRange(m_spellInfo, effIndex, minRadiusCaster, maxRadiusTarget);
+
+                // if we have a group, we don't care about other units around
+                if (group) {
```

Comment:
Whats the research on this? On classic chain heal can jump to an assistable NPC, which is for example Escort NPC. This makes it completely not work.

Reply:
Also, AnyFriendlyOrGroupMemberUnitInUnitRangeCheck  does all of this for us.